src/main/java/com/powsybl/openloadflow/util/BranchState.java

```diff
@@ -0,0 +1,42 @@
+package com.powsybl.openloadflow.util;
+
+import com.powsybl.openloadflow.network.*;
+
+import java.util.Collection;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+public class BranchState {
+    private final double a1;
+    private final double r1;
+
+    public BranchState(LfBranch b) {
+        PiModel piModel = b.getPiModel();
+        a1 = piModel.getA1();
+        r1 = piModel.getR1();
+    }
+
+    public void restoreBranchActiveState(LfBranch branch) {
```

Comment:
Can you here start with only `restoreBranchState` ? Sorry for the delay of the remark...